netty/src/main/java/io/grpc/transport/netty/BufferingHttp2ConnectionEncoder.java

```diff
@@ -67,6 +69,44 @@
  * in replacement for {@link io.netty.handler.codec.http2.DefaultHttp2ConnectionEncoder}.
  */
 class BufferingHttp2ConnectionEncoder extends DecoratingHttp2ConnectionEncoder {
+
+  /**
+   * Thrown if buffered streams are terminated due to this encoder being closed.
+   */
+  public static final class ChannelClosedException extends Http2Exception {
+    public ChannelClosedException() {
+      super(Http2Error.STREAM_CLOSED, "Connection closed");
+    }
+  }
+
+  /**
+   * Thrown if buffered streams are terminated due to receipt of a {@code GOAWAY}.
+   */
+  public static final class GoAwayException extends Http2Exception {
+    private final int lastStreamId;
+    private final long errorCode;
+    private final ByteBuf debugData;
```

Comment:
Yeah, probably should just convert to a UTF_8 string before creating the exception. WDYT?